ydb/core/tx/schemeshard/schemeshard__operation_alter_sequence.cpp

```diff
@@ -0,0 +1,486 @@
+
+using namespace NKikimr;
+using namespace NSchemeShard;
+
+class TConfigureParts: public TSubOperationState {
+private:
+    TOperationId OperationId;
+
+    TString DebugHint() const override {
+        return TStringBuilder()
+                << "TAlterSequence TConfigureParts"
+                << " operationId#" << OperationId;
+    }
+
+public:
+    TConfigureParts(TOperationId id)
+        : OperationId(id)
+    {
+        IgnoreMessages(DebugHint(), {TEvHive::TEvCreateTabletReply::EventType});
+    }
+
+    bool HandleReply(NSequenceShard::TEvSequenceShard::TEvUpdateSequenceResult::TPtr& ev,
+            TOperationContext& context) override {
+
+        auto ssId = context.SS->SelfTabletId();
+        auto tabletId = TTabletId(ev->Get()->Record.GetOrigin());
+        auto status = ev->Get()->Record.GetStatus();
+
+        LOG_DEBUG_S(context.Ctx, NKikimrServices::FLAT_TX_SCHEMESHARD,
+                    "TAlterSequence TConfigureParts HandleReply TEvUpdateSequenceResult"
+                    << " shardId# " << tabletId
+                    << " status# " << status
+                    << " operationId# " << OperationId
+                    << " at tablet " << ssId);
+
+        switch (status) {
+            case NKikimrTxSequenceShard::TEvUpdateSequenceResult::SUCCESS:
+                // Treat expected status as success
+                break;
+
+            default:
+                // Treat all other replies as unexpected and spurious
+                LOG_WARN_S(context.Ctx, NKikimrServices::FLAT_TX_SCHEMESHARD,
+                    "TAlterSequence TConfigureParts HandleReply ignoring unexpected TEvUpdateSequenceResult"
+                    << " shardId# " << tabletId
+                    << " status# " << status
+                    << " operationId# " << OperationId
+                    << " at tablet " << ssId);
+                return false;
+        }
+
+        TTxState* txState = context.SS->FindTx(OperationId);
+        Y_ABORT_UNLESS(txState);
+        Y_ABORT_UNLESS(txState->TxType == TTxState::TxAlterSequence);
+        Y_ABORT_UNLESS(txState->State == TTxState::ConfigureParts);
+
+        auto shardIdx = context.SS->MustGetShardIdx(tabletId);
+        if (!txState->ShardsInProgress.erase(shardIdx)) {
+            LOG_WARN_S(context.Ctx, NKikimrServices::FLAT_TX_SCHEMESHARD,
+                "TAlterSequence TConfigureParts HandleReply ignoring duplicate TEvUpdateSequenceResult"
+                << " shardId# " << tabletId
+                << " status# " << status
+                << " operationId# " << OperationId
+                << " at tablet " << ssId);
+            return false;
+        }
+
+        context.OnComplete.UnbindMsgFromPipe(OperationId, tabletId, txState->TargetPathId);
+
+        if (txState->ShardsInProgress.empty()) {
+            NIceDb::TNiceDb db(context.GetDB());
+            context.SS->ChangeTxState(db, OperationId, TTxState::Propose);
+            context.OnComplete.ActivateTx(OperationId);
```

Comment:
Call to `context.OnComplete.ActivateTx()` is unnecessary here -- on returning `true` (that is, on completing the state) it would be called automatically.